oadr/oadr/manager/eventmanager/EventManager.cpp

```diff
@@ -252,12 +266,19 @@ void EventManager::handleNewEvent(const string &eventID, const oadr2b::oadr::oad
 
 	m_events[eventID] = unique_ptr<oadr2b::oadr::oadrEvent>(eventCopy);
 
-	oadr2b::ei::OptTypeType::value optType = oadr2b::ei::OptTypeType::optOut;
+	if (oadr2b::oadr::ResponseRequiredType::value::always == event->oadrResponseRequired())
+	{
+		oadr2b::ei::OptTypeType::value optType = oadr2b::ei::OptTypeType::optOut;
 
-	m_service->OnEventNew(eventID, eventCopy, optType);
+		m_service->OnEventNew(eventID, eventCopy, optType);
 
-	Oadr2bHelper::appendEventResponse(eventResponses, "200", "OK", eventID,
-			eventCopy->eiEvent().eventDescriptor().modificationNumber(), optType, requestID);
+		Oadr2bHelper::appendEventResponse(eventResponses, "200", "OK", eventID,
+				eventCopy->eiEvent().eventDescriptor().modificationNumber(), optType, requestID);
+	}
+	else // oadr2b::oadr::ResponseRequiredType::value::never
+	{
+		m_service->OnEventNew(eventID, event);
```

Comment:
The wording of the conformance rule 12:
VEN, EiEvent Service, oadrCreatedEvent Payload

The VEN MUST respond with an oadrCreatedEvent to an event in oadrDistributeEvent based upon the value in each event’s oadrResponseRequired element as follows:

Always – The VEN MUST respond to the event with an oadrCreatedEvent eventResponse. This includes unchanged, new, changed, and canceled events.

Never – The VEN MUST NOT respond to the event with a oadrCreatedEvent eventResponse

Note that oadrCreatedEvent event responses SHOULD be returned in one message, but MAY be returned in separate messages.

Reply:
The wording of the conformance rule 62:
VEN, EiEvent Service, oadrDistributeEvent, oadrCreatedEvent Payload

The VEN MUST process every oadrEvent event message (new, modified, canceled, etc.) that it receives from the VTN in an oadrDistributeEvent payload and it MUST reply with a createdEvent message for every EiEvent message in which the responseRequired is set to “always”. Furthermore if the responseRequired is set to “never”, the VEN MUST NOT respond with a createdEvent message. It is at the complete discretion of the VTN as to whether responses are required from the VEN. Note that this rule is universal and applies to all scenarios including the following:

- The event is one in which the VEN is already aware.
- The event is being canceled and the VEN did not even know it existed
- It does not matter how the EiEvent payloads were delivered, i.e., PUSH, PULL or as the result of being delivered in an ALL payload